src/main/kotlin/com/ampnet/web3provider/service/impl/AccountInformationServiceImpl.kt

```diff
@@ -0,0 +1,57 @@
+package com.ampnet.web3provider.service.impl
+
+import com.ampnet.web3provider.controller.pojo.JsonRpcRequest
+import com.ampnet.web3provider.controller.pojo.ProviderResponse
+import com.ampnet.web3provider.enums.RedisEntity
+import com.ampnet.web3provider.repository.RedisRepository
+import com.ampnet.web3provider.service.AccountInformationService
+import com.ampnet.web3provider.service.DefaultProviderService
+import mu.KLogging
+import org.springframework.stereotype.Service
+
+@Service
+class AccountInformationServiceImpl(
+    private val redisRepository: RedisRepository,
+    private val defaultProviderService: DefaultProviderService
+) : AccountInformationService {
+
+    companion object : KLogging()
+
+    override fun getBalance(request: JsonRpcRequest): ProviderResponse {
+        val address = request.params[0]
+        val blockParameter = request.params[1]
```

Comment:
In none of the methods in this class you do a check for the length of the `params` list. If not enough params are provided, you would get `IndexOutOfBoundsException` which you did not handle explicitly. It would be good to check the length and then to throw an appropriate exception - `JsonRpcException` with an appropriate error response.

Reply:
Yeah sure, I haven't yet accounted for all sorts of exceptions in the code. I'll take care of it.